wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift

```diff
@@ -226,20 +230,37 @@ final class SettingsPropertyFactory {
         case .disableAnalyticsSharing:
             let getAction: GetAction = { [unowned self] _ in
                 if let tracking = self.tracking {
-                    return SettingsPropertyValue(tracking.disableAnalyticsSharing)
+                    return SettingsPropertyValue(tracking.isAnalyticsDisabled)
                 } else {
                     return SettingsPropertyValue(false)
                 }
             }
 
             let setAction: SetAction = { [unowned self] _, value, resultHandler in
-                if var tracking = self.tracking {
-                    switch value {
-                    case .number(let number):
-                        tracking.disableAnalyticsSharing(isDisabled: number.boolValue, resultHandler: resultHandler)
-                    default:
-                        throw SettingsPropertyError.WrongValue("Incorrect type \(value) for key \(propertyName)")
+                guard let tracking else {
+                    return
```

Comment:
Is this a valid state where `tracking` is nil?

Reply:
It's hard to say. The tracking manager is passed around as optional in a lot of places, my guess it because for testing a nil value is passed in. I would like to replace the settings system with something much much simpler, which is why I didn't worry too much about making this code nice.

Reply:
👍 Optionally consider logging an error if it is not there.